earn/src/data/MarginAccount.ts

```diff
@@ -111,95 +119,147 @@ export type UniswapPoolInfo = {
+        .div(10 ** token1.decimals)
+        .toNumber(),
+      uni0: Big(assetsData[4].toString())
+        .div(10 ** token0.decimals)
+        .toNumber(),
+      uni1: Big(assetsData[5].toString())
+        .div(10 ** token1.decimals)
+        .toNumber(),
+    };
+    const liabilities: Liabilities = {
+      amount0: Big(liabilitiesData[0].toString())
+        .div(10 ** token0.decimals)
+        .toNumber(),
+      amount1: Big(liabilitiesData[1].toString())
+        .div(10 ** token1.decimals)
+        .toNumber(),
+    };
+    marginAccounts.push({
+      address: accountAddress,
+      uniswapPool,
+      token0,
+      token1,
+      feeTier,
+      assets,
+      liabilities,
+      health: health.div(1e9).toNumber() / 1e9,
+    });
+  });
+  return marginAccounts;
 }
 
 export async function fetchMarketInfoFor(
   lenderLensContract: ethers.Contract,
   lender0: Address,
   lender1: Address
 ): Promise<MarketInfo> {
-  const [lender0Basics, lender1Basics] = await Promise.all([
-    lenderLensContract.readBasics(lender0),
-    lenderLensContract.readBasics(lender1),
-  ]);
+  const multicall = new Multicall({ ethersProvider: lenderLensContract.provider, tryAggregate: true });
+  const contractCallContext: ContractCallContext[] = [
+    {
+      reference: 'readBasics',
+      contractAddress: ALOE_II_KITTY_LENS_ADDRESS,
+      abi: KittyLensABI,
+      calls: [
+        {
+          reference: 'lender0',
+          methodName: 'readBasics',
+          methodParameters: [lender0],
+        },
+        {
+          reference: 'lender1',
+          methodName: 'readBasics',
+          methodParameters: [lender1],
+        },
+      ],
+    },
+  ];
 
-  const interestRate0 = new Big(lender0Basics.interestRate.toString());
+  const results = (await multicall.call(contractCallContext)).results;
+  const updatedReturnContext = convertBigNumbers(results['readBasics'].callsReturnContext);
+  const lender0Basics = updatedReturnContext[0].returnValues;
+  const lender1Basics = updatedReturnContext[1].returnValues;
+
+  const interestRate0 = new Big(lender0Basics[1].toString());
   const borrowAPR0 = interestRate0.eq('0') ? 0 : interestRate0.sub(1e12).div(1e12).toNumber() * secondsInYear;
-  const interestRate1 = new Big(lender1Basics.interestRate.toString());
+  const interestRate1 = new Big(lender1Basics[1].toString());
   const borrowAPR1 = interestRate1.eq('0') ? 0 : interestRate1.sub(1e12).div(1e12).toNumber() * secondsInYear;
-  const lender0Utilization = new Big(lender0Basics.utilization.toString()).div(10 ** 18).toNumber();
-  const lender1Utilization = new Big(lender1Basics.utilization.toString()).div(10 ** 18).toNumber();
-  const lender0TotalSupply = new Big(lender0Basics.totalSupply.toString());
-  const lender1TotalSupply = new Big(lender1Basics.totalSupply.toString());
-  const lender0TotalBorrows = new Big(lender0Basics.totalBorrows.toString());
-  const lender1TotalBorrows = new Big(lender1Basics.totalBorrows.toString());
```

Comment:
Looking at the documentation, it seems like these named outputs should still exist. Why did you switch to the index-based stuff below?

Reply:
Can you point me to where in the documentation you see this? I did not notice any named outputs for the results.

Reply:
<img width="709" alt="image" src="https://user-images.githubusercontent.com/17186559/219790447-e4c2e3cc-cb09-476c-8a6e-cc18770ba35e.png">

`returnValues` gives an object with an `amounts` property. That `amounts` label comes from the abi specified for the call.

Reply:
Maybe our abi is not getting passed in properly because I checked and there are no named outputs for the result when I do it.